include/nbl/asset/utils/ISPIRVDebloater.h

```diff
@@ -17,19 +17,31 @@ class ISPIRVDebloater final : public core::IReferenceCounted
 
         struct Result
         {
-            bool isAllEntryPointsFound;
             core::smart_refctd_ptr<ICPUBuffer> spirv; // nullptr if there is some entry point not found or spirv does not need to be debloated
+            bool isSuccess;
+
+            operator bool() const
+            {
+                return isSuccess;
+            }
         };
 
         struct EntryPoint
         {
-          std::string_view name;
-          hlsl::ShaderStage shaderStage;
-
-          bool operator==(const EntryPoint& rhs) const = default;
+            std::string_view name;
+            hlsl::ShaderStage shaderStage;
+
+            bool operator==(const EntryPoint& rhs) const = default;
+            auto operator<=>(const EntryPoint& other) const
+            {
+                if (auto cmp = shaderStage <=> other.shaderStage; cmp != 0)
+                    return cmp;
+                return name <=> other.name;
+            }
         };
 
-        Result debloat(const ICPUBuffer* spirvBuffer, std::span<const EntryPoint> entryPoints, system::logger_opt_ptr logger) const;
+        Result debloat(const ICPUBuffer* spirvBuffer, std::span<const EntryPoint> entryPoints, system::logger_opt_ptr logger = nullptr) const;
+        core::smart_refctd_ptr<ICPUBuffer> debloat(core::smart_refctd_ptr<ICPUBuffer>&& code, std::span<const EntryPoint> entryPoints, system::logger_opt_ptr logger = nullptr) const;
```

Comment:
take a `const ICPUBuffer` and return a `const ICPUBuffer`

Reply:
also it seems that it now makes most sense to take a `const core::set<EntryPoint>&` instead of a span